modules/drivers/bigquery-cloud-sdk/test/metabase/driver/bigquery_cloud_sdk_test.clj

```diff
@@ -371,6 +371,30 @@
           ;; make sure that the fake exception was thrown, and thus the query execution was retried
           (is (true? @fake-execute-called)))))))
 
+(deftest not-retry-cancellation-exception-test
+  (mt/test-driver :bigquery-cloud-sdk
+    (let [fake-execute-called (atom false)
+          orig-fn        @#'bigquery/execute-bigquery]
+      (testing "Should not retry query on cancellation"
+        (with-redefs [bigquery/execute-bigquery (fn [^BigQuery client ^String sql parameters _ _]
+                                                  ;; We dont want to retry the db setup queries, so only retry if the query contains "cancel_test"
+                                                  (if (and (re-find #"cancel_test" sql) (not @fake-execute-called))
+                                                    (do (reset! fake-execute-called true)
+                                                        ;; Simulate a CancellationException being thrown
+                                                        (throw (java.util.concurrent.CancellationException. "Query cancelled")))
```

Comment:
According to the suggestion, this would have to throw an `ex-info` with `::bigquery/cancelled? true`.

Reply:
fixed